test-utils/src/lib.rs

```diff
@@ -64,6 +65,18 @@ pub fn gen_text_with_non_ascii(c: u8, rng: &mut impl Rng, max_len: usize) -> Vec
     token_ticker
 }
 
+pub fn split_value(rng: &mut impl Rng, value: u128) -> Vec<u128> {
+    let mut numbers = vec![0, value];
+    let n = rng.gen_range(0..10);
+
+    if value > 1 && n > 0 {
+        numbers.extend((0..=n).map(|_| rng.gen_range(1..value)).collect::<Vec<_>>());
+        numbers.sort();
+    }
+
+    numbers.iter().tuple_windows().map(|(v0, v1)| v1 - v0).collect()
```

Comment:
Btw, there's a chance that the diffs are zero. I don't know if that's desired.

Reply:
I think that's ok